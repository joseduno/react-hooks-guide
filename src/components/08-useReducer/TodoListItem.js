import React from 'react'

const TodoListItem = ({todo, i, handleDone, handleDelete}) => {
    return (
        <li 
            className='list-group-item d-flex justify-content-between align-items-center'
        >
            <span
                type='button'
                className={todo.done ? 'text-decoration-line-through' : ''}
                onClick={() => handleDone(todo.id)}
            >
                {i+1}. {todo.desc}
            </span>
            <button 
                type='button'
                className='btn btn-danger btn-sm'
                onClick={()=>handleDelete(todo.id)}
            >
                Borrar
            </button>
        </li>
    )
}

export default TodoListItem
