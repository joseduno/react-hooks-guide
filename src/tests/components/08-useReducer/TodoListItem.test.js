import { demoTodos } from "../../fixtures/demoTodos"
import TodoListItem from '../../../components/08-useReducer/TodoListItem'
import { shallow } from "enzyme";


describe('Pruebas en TodoListItem', () => {

    const todo = demoTodos[0]
    const index = 0
    const handleDone = jest.fn()
    const handleDelete = jest.fn()

    const wrapper = shallow(<TodoListItem 
        todo={todo}
        i={index}
        handleDone={handleDone}
        handleDelete={handleDelete}
    />)

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('debe llamar la funcion borrar', () => {
        wrapper.find('button').simulate('click')
        expect(handleDelete).toHaveBeenCalledWith(todo.id)
    })

    test('debe llamar la funcion toggle', () => {
        wrapper.find('span').simulate('click')
        expect(handleDone).toHaveBeenCalledWith(todo.id)
    })
    
    test('debe mostrar el texto correctamente ', () => {
        const span = wrapper.find('span')
        expect(span.text().trim()).toBe(`${0+1}. ${todo.desc}`)
    })
    
    test('debe mostrar la clase text-decoration-line-through en la etiqueta <span></span>', () => {
        let todo = demoTodos[0]
        todo.done = true
        const index = 0
        const handleDone = jest.fn()
        const handleDelete = jest.fn()
        const wrapper = shallow(<TodoListItem 
            todo={todo}
            i={index}
            handleDone={handleDone}
            handleDelete={handleDelete}
        />)
        expect(wrapper.find('span').hasClass('text-decoration-line-through')).toBe(true)
    })
    
})
