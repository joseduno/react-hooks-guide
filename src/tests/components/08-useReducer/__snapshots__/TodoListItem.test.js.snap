// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en TodoListItem debe mostrarse correctamente 1`] = `
<li
  className="list-group-item d-flex justify-content-between align-items-center"
>
  <span
    className=""
    onClick={[Function]}
    type="button"
  >
    1
    . 
    Aprender React
  </span>
  <button
    className="btn btn-danger btn-sm"
    onClick={[Function]}
    type="button"
  >
    Borrar
  </button>
</li>
`;
